### ISSUE TITLE
[no-unused-vars] enums are ignored

### ISSUE THREAD

**Issue by avalanche1** [typescript-eslint/typescript-eslint#3080]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**
```ts
export const enum Config {
  ABC = "1.2",
}
```
`ABC` is not reported as unused.

```yaml
root: true
settings: {}

parser: "@typescript-eslint/parser"
parserOptions:
  ecmaVersion: 2020
  project: "./tsconfig.json"
  sourceType: module

env:
  es2020: true
  browser: true

extends:
  - "eslint:recommended"
  - "plugin:@typescript-eslint/eslint-recommended"
  - "plugin:@typescript-eslint/all"
  - "plugin:functional/all"
  - "plugin:jest/all"
  - "plugin:jest-dom/recommended"
  - "plugin:no-unsanitized/DOM"
  - "plugin:promise/recommended"
  - "plugin:react/all"
  - "plugin:testing-library/recommended"
  - "prettier/@typescript-eslint"
  - "prettier"
plugins:
  - "@typescript-eslint"
  - functional
  - jest
  - jest-dom
  - no-unsanitized
  - promise
  - react
  - react-hooks
  - testing-library

rules: 
  "@typescript-eslint/no-unused-vars": warn
  no-unused-vars: off
```

`tsconfig`
```ts
{
  "compilerOptions": {
    "allowJs": true,
    "allowSyntheticDefaultImports": true,
    "alwaysStrict": true,
    "forceConsistentCasingInFileNames": true,
    "isolatedModules": true,
    "jsx": "react",
    "esModuleInterop": true,
    "lib": ["es2020", "dom", "dom.iterable"],
    "module": "esnext",
    "moduleResolution": "node",
    "noEmit": true,
    "noImplicitReturns": true,
    "noUncheckedIndexedAccess": true,
    "resolveJsonModule": true,
    "skipLibCheck": true,
    "sourceMap": true,
    "strict": true,
    "suppressImplicitAnyIndexErrors": true,
    "target": "es2018",
    "typeRoots": ["node_modules/@types"],
    "types": ["jest", "expect-more-jest"]
  },
  "include": [
    "globals.d.ts",
    "assets/**/*",
    "src/**/*.ts",
    "src/**/*.tsx",
    "tests/**/*.ts",
    "tests/**/*.tsx",
    "utility-belt/**/*.ts",
    "utility-belt/**/*.tsx"
  ]
}
```

**Expected Result**
`ABC` to be reported as a warning.

**Actual Result**
`ABC` is ignored.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `^4.15.1` |
| `@typescript-eslint/parser`        | `^4.15.1` |
| `TypeScript`                       | `^4.1.3` |
| `ESLint`                           | `^7.20.0` |

**Comment by avalanche1** [comment]
Hmm... you are right.
Would you suggest a rule to check for object members?

**Comment by bradzacher** [comment]
There isn't one.

Detecting member usage is difficult to impossible to do, due to JS allowing you to access members dynamically.
